src/hotspot/share/nmt/memTracker.cpp

```diff
@@ -54,6 +54,8 @@ MemBaseline MemTracker::_baseline;
 bool MemTracker::NmtVirtualMemoryLocker::_safe_to_use;
 
 void MemTracker::initialize() {
+  NMT_LogRecorder::initialize();
```

Comment:
Why we initialize the log-recorder even if NMT is not enabled?

Reply:
We allow the NMTBenchmark to run without NMT, so that we can compare memory usage with NMT on or off.

NMTBenchmark doesn't require NMT to be ON.

Now that you brought it up, the name might be confusing , but the main usage was to compare apps with NMT on and off.